src/main/java/nl/tudelft/lifetiles/traverser/models/ReferencePositionTraverser.java

```diff
@@ -0,0 +1,129 @@
+package nl.tudelft.lifetiles.traverser.models;
+
+import nl.tudelft.lifetiles.graph.models.Edge;
+import nl.tudelft.lifetiles.graph.models.Graph;
+import nl.tudelft.lifetiles.graph.models.sequence.SegmentString;
+import nl.tudelft.lifetiles.graph.models.sequence.Sequence;
+import nl.tudelft.lifetiles.graph.models.sequence.SequenceSegment;
+
+/**
+ * Traverser which generates for each segment the coordinates in comparison with
+ * the reference sequence.
+ *
+ * @author Jos
+ *
+ */
+public class ReferencePositionTraverser {
+
+    /**
+     * Reference sequence which segments are compared to.
+     */
+    private Sequence referenceVar;
+
+    /**
+     * Constructs a ReferencePositionTraverser.
+     *
+     * @param reference
+     *            Reference sequence which segments are compared to.
+     */
+    public ReferencePositionTraverser(final Sequence reference) {
+        referenceVar = reference;
+    }
+
+    /**
+     * Copy of the graph to be traversed.
+     */
+    private Graph<SequenceSegment> graphVar;
+
+    /**
+     * Traverses and copies the graph.
+     *
+     * @param graph
+     *            Graph to be copied and traversed.
+     * @return Traversed copy of the graph with reference positions.
+     */
+    public final Graph<SequenceSegment> traverseGraph(
+            final Graph<SequenceSegment> graph) {
+        graphVar = graph.copy();
```

Comment:
The graph should not be copied inside the constructor but before the call to it.